dotCMS/src/main/java/com/dotcms/graphql/business/RelationshipFieldGenerator.java

```diff
@@ -0,0 +1,122 @@
+package com.dotcms.graphql.business;
+
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;
+import static graphql.schema.GraphQLList.list;
+import static graphql.schema.GraphQLNonNull.nonNull;
+
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.graphql.datafetcher.RelationshipFieldDataFetcher;
+import com.dotcms.graphql.exception.FieldGenerationException;
+import com.dotcms.util.DotPreconditions;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.RelationshipAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.structure.model.ContentletRelationships;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLTypeReference;
+
+/**
+ * This implementation generates a {@link GraphQLFieldDefinition} only for {@link Field}s
+ * of type {@link com.dotcms.contenttype.model.field.RelationshipField}
+ */
+class RelationshipFieldGenerator implements GraphQLFieldGenerator {
+
+    private RelationshipAPI relationshipAPI = APILocator.getRelationshipAPI();
+
+    @VisibleForTesting
+    protected void setRelationshipAPI(RelationshipAPI relationshipAPI) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'relationshipAPI' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)